hackathon_site/hardware/api_filters.py

```diff
@@ -0,0 +1,25 @@
+from django_filters import rest_framework as filters
+from django_filters.fields import Lookup
+
+from hardware.models import Hardware
+from hardware.serializers import HardwareSerializer
+
+
+class HardwareFilter(filters.FilterSet):
+    queryset = Hardware
+    serializer_class = HardwareSerializer
+
+    class Meta:
+        model = Hardware
+        fields = ["name", "quantity_available"]
+
+    in_stock = filters.BooleanFilter(label="In stock?", method="filter_in_stock")
+
+    @staticmethod
+    def filter_in_stock(queryset, _, value):
+        if value is True:
+            return queryset.filter(quantity_available__gt=0)
+        else:
+            return queryset.filter(quantity_available__lte=0)
+
+    id = filters.BaseInFilter(field_name="id", label="Id(s) of the hardware item")
```

Comment:
There's an edge case here, if you submit something that's a non-integer (eg `id=asdf`), the page will crash.

To fix that, we need to add a validator for the field, and throw a `400 Bad Request` response if the user gives bad IDs (crashes are always bad). Validators need to be set on the actual field inside the filter, but I can't quite figure out how to do that yet. Will report back

Reply:
Hey Graham, I found one solution but this is not the most ideal one.

![image](https://user-images.githubusercontent.com/36619969/125758841-c4c2288c-09a1-4159-8b10-15e7b99810f2.png)

But I can input them in the address bar.
![image](https://user-images.githubusercontent.com/36619969/125759376-eb621fdd-cab6-4bc7-8a58-cb0cf77a4b1d.png)

When I input characters other than numbers, the error will be thrown out.

![image](https://user-images.githubusercontent.com/36619969/125759465-9f5a2197-82c2-4cbe-98c9-b3c2137a4458.png)

@grahamhoyes